From e64657055a410ec7e8caa251ee7ec42f838fe8ae Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Fri, 5 Sep 2014 01:49:22 -0400
Subject: [iommu] vt-d: Simplify iommu check in domain_remove_one_dev_info()

Message-id: <20140905014922.31297.9080.stgit@gir.stowe>
Patchwork-id: 91329
O-Subject: [RHEL7.1 PATCH 36/65] iommu/vt-d: Simplify iommu check in domain_remove_one_dev_info()
Bugzilla: 1129880 1087643
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1087643
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1129880
Upstream Status: 8bbc4410129c0919e5943012489427c5e050a63e

commit 8bbc4410129c0919e5943012489427c5e050a63e
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Sun Mar 9 13:52:37 2014 -0700

    iommu/vt-d: Simplify iommu check in domain_remove_one_dev_info()

    Now we store the iommu in the device_domain_info, we don't need to do a
    lookup.

    Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/intel-iommu.c b/drivers/iommu/intel-iommu.c
index 8303f25..dc322d0 100644
--- a/drivers/iommu/intel-iommu.c
+++ b/drivers/iommu/intel-iommu.c
@@ -4056,8 +4056,7 @@ static void domain_remove_one_dev_info(struct dmar_domain *domain,
    * owned by this domain, clear this iommu in iommu_bmp
    * update iommu count and coherency
    */
-  if (iommu == device_to_iommu(info->segment, info->bus,
-         info->devfn))
+  if (info->iommu == iommu)
    found = 1;
  }
 
-- 
1.7.1